src/map/lua/lua_baseentity.cpp

```diff
@@ -664,6 +664,28 @@ void CLuaBaseEntity::setVolatileCharVar(std::string const& varName, int32 value,
     }
 }
 
+/************************************************************************
+ *  Function: getAllLocalVars()
+ *  Purpose : Returns all variables assigned locally to an entity
+ *  Example :
+ *  Notes   :
+ ************************************************************************/
+
+auto CLuaBaseEntity::getAllLocalVars() -> sol::table
+{
+    auto                          table     = lua.create_table();
+    std::map<std::string, uint32> localVars = m_PBaseEntity->GetAllLocalVars();
+
+    for (auto var : localVars)
```

Comment:
This creates copies and throws them away on every iteration of this loop.

`for (auto const& [varname, value] : ...`

Reply:
> This creates copies and throws them away on every iteration of this loop.
> 
> `for (auto const& [varname, value] : ...`

Thanks, I updated the PR, hopefully this is ok now.